# ğŸ³ Snakemake + Docker Pipeline for Proteomics Analysis

This repository contains the code and data to reproduce the analysis from an unpublished study on Î²2-agonist treatment combined with resistance training and their effects on the skeletal muscle proteome in
type I and IIa muscle fibers, respectively.

The repository is designed to be executed as a **Snakemake workflow** inside a **Docker container**. Inside the docker container, [`renv`](https://rstudio.github.io/renv/) manages R package dependencies. In this way, the project is reproducible and easily useable across systems. 

You do **not need to install R or any R packages** locally â€” everything is self-contained within the Docker image.

## ğŸš€ Quick Start (Run the Analysis in Docker)

To reproduce the full analysis in a portable and reproducible environment:

### 1. ğŸ“¥ Clone the repository
```bash
git clone https://github.com/s-jessen/manuscript_beta2_single_fiber.git
cd manuscript_beta2_single_fiber
```

### 2. ğŸ³ Install Docker (if not already installed)
```
#Linux:
sudo pacman -S docker
sudo systemctl start docker
```
On macOS or Windows, download from [`Docker website`](https://www.docker.com/products/docker-desktop/)

### 3. ğŸ“¦ Download the docker image
```
docker pull sorenjessen/b2a_res_single_fiber:latest
```

### 4. â–¶ï¸ Run the analysis from within the root of the repository

This command will mount your local 'raw-data' (input data), 'R' (scripts), and 'data' (output files such as Figures etc.) folders into the container and run the full pipeline:
```
## ----- On linux/mac ----- ##
bash run_docker_unix.sh

## ----- On Windows ----- ##
#You may need to run PowerShell as admin and run
Set-ExecutionPolicy -Scope Process -ExecutionPolicy Bypass

#...followed by
./run_docker_win.ps1

```

This will automatically:

- Mount the necessary folders (data-raw, R, data)
- Run the Snakemake workflow with 1 fewer core than your systemâ€™s total
- Write the output to the local data/ folder

You can modify the analysis scripts in the local R/ folder and rerun the command. The pipeline will automatically rerun only the necessary steps based on file dependencies.

![DAG](workflow.svg)

### ğŸ“ Project Folder Structure

This project follows a reproducible research structure using the `prodigenr` framework. Below is an overview of the folder layout and purpose of key files:

```
â”œâ”€â”€ data                                    # Output of scripts (GSEA results, statistics, etc.)
â”‚   â”œâ”€â”€ figures                             # Folder to store figures
â”‚   â””â”€â”€ ...                                 # RDS files, tables, etc.
â”œâ”€â”€ data-raw                                # Raw input data
â”‚   â”œâ”€â”€ design.xlsx                         # Metadata file
â”‚   â”œâ”€â”€ keywords.xlsx                       # Uniprot keywords for gene annotation
â”‚   â”œâ”€â”€ mitocarta.xls                       # MitoCarta 3.0 gene annotation
â”‚   â””â”€â”€ proteomic_data.csv                  # Raw proteomic intensity data
â”œâ”€â”€ DESCRIPTION                             # Metadata and dependencies for R package structure
â”œâ”€â”€ Dockerfile                              # Docker recipe to build reproducible container
â”œâ”€â”€ docs                                    # Manuscript figures
â”œâ”€â”€ environment.yml                         # Conda environment file (for Snakemake, Python tools)
â”œâ”€â”€ manuscript_beta2_single_fiber_snakemake.Rproj # R project file (not necessary for Docker workflow)
â”œâ”€â”€ R                                       # R analysis scripts
â”‚Â Â  â”œâ”€â”€ data_preparation.R                  # Cleans raw data and stores SummarizedExperiment
â”‚Â Â  â”œâ”€â”€ figure1.R                           # Code producing figure 1 plots
â”‚Â Â  â”œâ”€â”€ figure2.R                           # Code producing figure 2 plots
â”‚Â Â  â”œâ”€â”€ figure3.R                           # Code producing figure 3 plots
â”‚Â Â  â”œâ”€â”€ functions.R                         # Functions
â”‚Â Â  â”œâ”€â”€ gsea_interaction.R                  # Runs GSEA (gene set enrichment analysis) for interaction results
â”‚Â Â  â”œâ”€â”€ gsea.R                              # Runs GSEA 
â”‚Â Â  â”œâ”€â”€ limma.R                             # Differential expression using limma
â”‚Â Â  â””â”€â”€ settings.R                          # Plotting themes, color palettes, etc.
â”œâ”€â”€ README.md                               # This file.
â”œâ”€â”€ renv                                    # renv infrastructure
â”‚Â Â  â””â”€â”€ ...
â”œâ”€â”€ renv.lock                               # Exact R package versions
â”œâ”€â”€ run_in_docker_container.sh              # Bash script initiating snakemake pipeline in Docker container
â”œâ”€â”€ snakefile                               # Bash script to launch pipeline in Docker
â””â”€â”€ workflow.svg                            # DAG of pipeline
```

The R project contains code to reproduce the following highlighted figures:

![Figure 1](docs/Figure1.png)
![Figure 2](docs/Figure2.png)
![Figure 3](docs/Figure3.png)
![Figure 4](docs/Figure4.png)
![Figure 5](docs/Figure5.png)
![Figure 6](docs/Figure6.png)
